### DISCUSSION TITLE
Should we localise base celebration names when we use only one with a patronage title?

### DISCUSSION THREAD

**Discussion by tukusejssirs** [romcal/romcal#453] [Q&A]
I have just noticed that #431 introduced a potential anomaly.

`pt-br` is the only locale that has `our_lady_of_aparecida` key localised. That key is not used in any celebration. It is only used in `martyrology.ts`, however, AFAIK it does not need to be localised.

```bash
# grep -r our_lady_of_aparecida lib/
lib/locales/la.ts:    our_lady_of_aparecida_patroness_of_brazil: 'B. Mariæ Virginis de Aparecida, Patronæ Brasiliæ',
lib/locales/cs.ts:    our_lady_of_aparecida_patroness_of_brazil: 'Panny Marie z Aparecidy, patronky Brazílie',
lib/locales/es.ts:    our_lady_of_aparecida_patroness_of_brazil: 'Nuestra Señora de Aparecida, Patrona de Brasil',
lib/locales/sk.ts:    our_lady_of_aparecida_patroness_of_brazil: 'Preblahoslavenej Panny Márie z Aparecidy, patrónky Brazílie',
lib/locales/en.ts:    our_lady_of_aparecida_patroness_of_brazil: 'Our Lady of Aparecida, Patroness of Brazil',
lib/locales/pt-br.ts:    our_lady_of_aparecida: 'Nossa Senhora da Conceição Aparecida',
lib/locales/pt-br.ts:    our_lady_of_aparecida_patroness_of_brazil: 'Nossa Senhora Aparecida',
lib/particular-calendars/brazil.ts:    our_lady_of_aparecida: {
lib/particular-calendars/brazil.ts:      customLocaleId: 'our_lady_of_aparecida_patroness_of_brazil',
lib/catalog/martyrology.ts:    our_lady_of_aparecida: {
```

Now, the question is: should we localise such keys or not? If yes, we should localise other keys where only a key with a patronage title is used. If not, we should remove this key from `pt-br`. :thinking:

WDYT, @emagnier? :pray:

I can do it, I just want to hear you opinion first. :pray:

**Comment by tukusejssirs** [comment]
Same issue is with `andre_bauer_martyr`, although there are multiple issues with this key:

- `modestus_andlauer_and_andrew_bauer_martyrs` is only used in `france.strasbourg`;
- `andre_bauer_martyr` is not used anywhere (I forgot to rename it in one of my PRs);
- `andrew_bauer_martyr` is used in `martyrology.ts` only.

Therefore, IMHO we should remove `andrew_bauer_martyr` from `fr`.

WDYT? :thinking:

**Comment by tukusejssirs** [comment]
Same issue is with `our_lady_of_lujan`: only `our_lady_of_lujan_patroness_of_argentina` is used in `argentina`, therefore I suggest to remove `our_lady_of_lujan`.

**Reply by emagnier** [reply]
I have just provided some responses here on this topic: https://github.com/romcal/romcal/issues/136#issuecomment-1701486145

However, for the case of `our_lady_of_lujan` (and similar cases), both translation keys, `our_lady_of_lujan` and `our_lady_of_lujan_patroness_of_argentina`, need to be kept.

This is because the translation key `our_lady_of_lujan_patroness_of_argentina` only applies to the `argentina.ts` calendar. In other calendars where this celebration is observed, the translation `key our_lady_of_lujan` continues to be used.

In the `argentina.ts` calendar, it can be observed that the `our_lady_of_lujan` definition is used in conjunction with a `customLocaleId` property. This avoids redefining this celebration in that calendar, especially in cases where this celebration might be already defined from an inherited calendar (not the case here, but apply for example to `benedict_of_nursia_abbot` in `europe.ts`), and also avoids redefining the corresponding entry in the martyrology. It's the same entity but with a specific translation for that particular calendar, and potentially other properties redefined, like the date, the rank, the titles... In this case of `argentina.ts`, you will note that a `titles` property is adding a specific title for Argentina (in addition to those potentially defined in the martyrology).

You have quite a similar situation where `andre_bauer_martyr` is used only in the martyrology, and `modestus_andlauer_and_andrew_bauer_martyrs` is defined in a calendar. These 2 keys must also be kept, and ideally translated in all languages.
